tests/unit/s2n_locking_test.c

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "s2n_test.h"
+#include <pthread.h>
+
+#include "crypto/s2n_locking.h"
+
+#define LOCK_N 1
+
+#if !(S2N_OPENSSL_VERSION_AT_LEAST(1, 1, 0))
+
+static void* s2n_test_thread(void *arg)
+{
+    bool *passed_lock = (bool*) arg;
+    CRYPTO_lock(CRYPTO_LOCK, LOCK_N, NULL, 0);
+    *passed_lock = true;
+    CRYPTO_lock(CRYPTO_UNLOCK, LOCK_N, NULL, 0);
+    return NULL;
+}
+
+static void s2n_test_locking_cb(int mode, int n, char *file, int line)
+{
+    return;
```

Comment:
You could, potentially, do some counting in here to detect you have visited the function.

Reply:
This function is never actually visited. I just use it in the test of init / cleanup. It just needs to exist.

Reply:
Uhh, we are missing a good test case then :)

Reply:
What would the missing test case be?

Reply:
We think we already test this by checking locking is enabled and the return values of the getter function.